R/helpers.R

```diff
@@ -589,3 +589,41 @@ h_plot_data_df <- function(data, blind = FALSE, ...) {
 
   df
 }
+
+#' Prepare Cohort Lines for Data Plot
+#'
+#' @description `r lifecycle::badge("experimental")`
+#'
+#' This helper function prepares a `ggplot` geom with reference lines
+#' separating different cohorts on the plot of `Data` class object.
+#' Lines are either vertical or horizontal of green color and longdash type.
+#'
+#' @details The geom object is returned if and only if `placebo` is equal to `TRUE`
+#'   and there are more than one unique values in `cohort`. Otherwise, this function
+#'   returns `NULL` object.
+#'
+#' @param cohort (`integer`)\cr the cohort indices.
+#' @param vertical (`flag`)\cr should the line be vertical? Otherwise it is horizontal.
+#' @param placebo (`flag`)\cr is placebo included in the doses?
+#'   If it so, this function returns `NULL` object as in this case all doses
+#'   in a given cohort are equal and there is no need to separate them.
+#'
+h_plot_data_cohort_lines <- function(cohort,
+                                     vertical = TRUE,
+                                     placebo) {
```

Comment:
better the other way around - arguments that have defaults should be at the end of the function signature